File: DSA Learning Series/Contest 1 - Complexity Analysis and Basic Warm Up/Laddu/CPP/laddu.cpp
// Author - Shreyas Chavhan
// Github - https://www.github.com/shreyaschavhan

#include <bits/stdc++.h>
using namespace std;

int main(){
    ios_base :: sync_with_stdio(false);
    cin.tie(NULL);

    int T;
    cin >> T;
    while(T--){
        int activities;
        string origin;
        cin >> activities >> origin;
        int LADDU = 0;
        while(activities--){
            string activity;
            cin >> activity;
            if(!activity.compare("CONTEST_WON")){
                int rank;
                cin >> rank;
                LADDU += 300;
                if(rank < 20){
                    LADDU += (20 - rank);
                }
            }
            if(!activity.compare("TOP_CONTRIBUTOR")){
                LADDU += 300;
            }
            if(!activity.compare("BUG_FOUND")){
                int severity;
                cin >> severity;
                LADDU += severity;
            }
            if(!activity.compare("CONTEST_HOSTED")){
                LADDU += 50;
            }
        }
        if(origin == "INDIAN"){
            cout << LADDU/200 << '\n';
        }
        else{
            cout << LADDU/400 << '\n';
        }
    }
    return 0;
}

File: DSA Learning Series/Contest 1 - Complexity Analysis and Basic Warm Up/Laddu/PYTHON/Laddu.py
"""
    AUTHOR  -  Atharva Deshpande
    GITHUB  -  https://github.com/AtharvaD11
    QUESTION LINK  -  https://www.codechef.com/LRNDSA01/problems/LADDU
"""

*************************************************************

for _ in range(int(input())):
    n_nat = input().split()
    lst = []
    for i in range(int(n_nat[0])):
        lst.append(input().split())
    laddoos = 0
    for i in lst:
        if i[0] == 'CONTEST_WON':
            if int(i[1]) <=20:
                laddoos += (300 + (20-int(i[1])))
            else:
                laddoos += 300
        if i[0] == 'TOP_CONTRIBUTOR':
            laddoos += 300
        if i[0] == 'BUG_FOUND':
            laddoos += (int(i[1]))
        if i[0] == 'CONTEST_HOSTED':
            laddoos += 50
    if n_nat[1] == 'INDIAN':
        print(laddoos//200)
    else:
        print(laddoos//400)

